### PR TITLE
[Benchmarks] Add optional performance transition tests for Reply

### PR DESCRIPTION
**Pull request by Renzo-Olivares**
This change adds optional performance transitions test similar to what was done in Crane here https://github.com/flutter/gallery/pull/216 . 

This tests goes through the flow of Reply, going through each area where this is a potential for a performance drop , travelling through each of the motion transitions Reply has.

To run the transitions_perf test for all demos:
`flutter drive --profile --trace-startup -t test_driver/transitions_perf.dart -d <device>`
To run the transitions_perf test for just Reply:
`flutter drive --profile --trace-startup -t test_driver/transitions_perf.dart -d <device> --dart-define=onlyReply=true`

Sample output:

```
average_frame_build_time_millis: 2.830270543615678, 
90th_percentile_frame_build_time_millis: 4.346, 
99th_percentile_frame_build_time_millis: 25.74, 
worst_frame_build_time_millis: 72.224, 
missed_frame_build_budget_count: 15, 
average_frame_rasterizer_time_millis: 9.613210858585852, 
90th_percentile_frame_rasterizer_time_millis: 12.598, 
99th_percentile_frame_rasterizer_time_millis: 38.464, 
worst_frame_rasterizer_time_millis: 223.061, 
missed_frame_rasterizer_budget_count: 38, 
frame_count: 791, 
frame_rasterizer_count: 792, 
average_vsync_transitions_missed: 1.209125475285171, 
90th_percentile_vsync_transitions_missed: 2.0, 
99th_percentile_vsync_transitions_missed: 4.0
```

Looks like there could be room for improvement here, being our frames are out of budget in our build and rasterizer.

Note: Maybe these performance transitions tests could also be useful for the motion demos

### CODE REVIEW

File: lib/studies/reply/adaptive_nav.dart [hunk 1]
```diff
@@ -1207,6 +1211,7 @@ class _ReplyFabState extends State<_ReplyFab>
               return Tooltip(
                 message: tooltip,
                 child: InkWell(
+                  key: const ValueKey('ReplyFAB'),
```
Comment (clocksmith) [inline lib/studies/reply/adaptive_nav.dart:1214]: nit: ReplyFab?

File: test_driver/transitions_perf_test.dart [hunk 2]
```diff
@@ -250,8 +264,42 @@ void main([List<String> args = const <String>[]]) {
...
+            driver,
+            additionalActions: () async {
+              await driver.tap(replyFab); //tap compose
+              await driver.tap(replyExit); //exit compose page
+              await driver.tap(replySearch); //tap search icon
+              await driver.tap(replyExit); //exit search page
+              await driver.tap(replyEmail); //tap email card
+              await driver.tap(replyExit); //exit email card
+              await driver.tap(replyLogo); //tap reply logo
+              await driver.tap(replyLogo); //tap reply logo again
```
Comment (clocksmith) [inline test_driver/transitions_perf_test.dart:284]: mention that this will expand the drawer/rail?

File: test_driver/transitions_perf_test.dart [hunk 3]
```diff
@@ -250,8 +264,42 @@ void main([List<String> args = const <String>[]]) {
...
+      if (!isTestingReplyOnly) return;
+
+      // Collect timeline data for just the Crane study.
+      final timeline = await driver.traceAction(
+        () async {
+          await runDemos(
+            ['reply@study'],
+            driver,
+            additionalActions: () async {
+              await driver.tap(replyFab); //tap compose
```
Comment (clocksmith) [inline test_driver/transitions_perf_test.dart:277]: nit: Use regular punctuation for these comments

File: test_driver/transitions_perf_test.dart (hunk 2 above)
Comment (Renzo-Olivares) [reply test_driver/transitions_perf_test.dart:284]: Thanks for reminding me! I forgot to test the rail. I will do that too.

**Review by clocksmith — APPROVED** [review]

**Review by guidezpl — APPROVED** [review]
Nice!

```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Pull request merged**